exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinExporterConfiguration.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.zipkin;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.internal.Utils;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * Configurations for {@link ZipkinSpanExporter}.
+ *
+ * @since 0.4.0
+ */
+@AutoValue
+@Immutable
+public abstract class ZipkinExporterConfiguration {
```

Comment:
consider adding a static factory method to simplify this.

ex
```java
  /**
   * Builds a HTTP exporter for <a href="https://zipkin.io/zipkin-api/#/">Zipkin V2</a> format.
   * 
   * @param endpoint See {@link #getEndpoint()}, ex. "http://zipkinhost:9411/api/v2/spans".
   */
  public static ZipkinExporterConfiguration create(String endpoint) {
... newbuilder.endpoint().build()
```

Reply:
In fact.. if you do this, you can probably get out of the complex statements in the builder. remove the "endpoint" variant, and require anyone using the builder to supply their own sender and encoder. SOOO much simpler.

Reply:
would you have an overload for the `serviceName`, or just require people who want to set that to use the builder directly?

Reply:
I tried this out, including the `serviceName` option. Let me know what you think.

Reply:
I guess the local service name is still not a otel concept right?

Reply:
I think(?) this is the `service.name` resource attribute: https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#service

Reply:
If this is the same thing as the Resource `service.name`, then that is a property that is available to the exporter via the Resource attached to the `SpanData` instance. It's not generally something you configure at the exporter level, at least not from what I've seen so far.